content/docs/user-guide/experiment-management/checkpoints.md

```diff
@@ -0,0 +1,454 @@
+to save the model files at certain points during a training process.
+
+With DVC experiments and checkpoints, you can:
+
+- Implement the best practice in deep learning to save your model weights as
+  checkpoints.
+- Track all code and data changes corresponded to the checkpoints.
+- See when metrics start diverging and revert to the optimal checkpoint.
+- Automate the process of tracking every training epoch.
+
+[The way checkpoints are implemented by DVC](/blog/experiment-refs) utilizes
+_ephemeral_ experiment commits and experiment branches within DVC. They are
+created using the metadata from experiments and are tracked with the `exps`
+custom Git reference.
+
+You can add experiments to your Git history by committing the experiment you
+want to track, which you'll see later in this tutorial.
+
+## Setting up the project
+
+This tutorial is going to cover how to implement checkpoints in a ML project
+using DVC. We're going to train a model to identify handwritten digits based on
+the MNIST dataset.
+
+You can follow along with the steps here or you can clone the repo directly from
+GitHub and play with it. To clone the repo, run the following commands.
+
+```bash
+git clone https://github.com/iterative/checkpoints-tutorial
+cd checkpoints-tutorial
+```
+
+It is highly recommended you create a virtual environment for this example. You
+can do that by running:
+
+```bash
+python3 -m venv .venv
+```
+
+Once your virtual environment is installed, you can start it up with one of the
+following commands.
+
+- On Mac/Linux: `source .venv/bin/activate`
+- On Windows: `.\.venv\Scripts\activate`
+
+Once you have your environment set up, you can install the dependencies by
+running:
+
+```bash
+pip install -r requirements.txt
+```
+
+This will download all of the packages you need to run the example. Now you have
+everything you need to get started with experiments and checkpoints.
+
+## Setting up a DVC pipeline
+
+DVC versions data and models and it also can version the machine learning model
+weights file, the checkpoints, during the training process. To enable this, you
+will need to run your model training under DVC pipeline.
+
+Adding DVC pipline only takes a few commands. At the root of the project, run:
+
+```bash
+dvc init
+```
+
+This sets up the files you need for your DVC pipeline to work.
+
+This will create a _dvc.yaml_ file in your root directory.
+
+Now we need to add a stage for training our model within a DVC pipeliene. We'll
+do that with `dvc stage add`, which we'll explain more later. For now, run the
+following command:
+
+```bash
+dvc stage add -n train -d data/MNIST -d train.py \
+              -c model.pt --plots-no-cache predictions.json \
+              -p seed,lr,weight_decay --live dvclive python train.py
+```
+
+The checkpoints need to be enabled in DVC at the pipeline level. The
+`-c / --checkpoint` option of the `dvc stage add` command defines the checkpoint
+file or directory.
```

Comment:
I'm not sure it's clear why `model.pt` is the checkpoint file here and not `predictions.json` or some other output. It might be helpful to clarify that the checkpoint file is an output from one checkpoint that becomes a dependency for the next checkpoint, such as the model weights file. cc @jorgeorpinel @pmrowla

Reply:
Agree, introduce "circular dependency" concept (not sure about the terminology though)